autoblocks/_impl/tracerv2/run.py

```diff
@@ -0,0 +1,263 @@
+import asyncio
+import contextvars
+import datetime
+import inspect
+import logging
+import threading
+from datetime import timedelta
+from typing import Any
+from typing import Coroutine
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import httpx
+
+from autoblocks._impl.config.constants import INGESTION_ENDPOINT
+from autoblocks._impl.testing.models import BaseEvaluator
+from autoblocks._impl.testing.models import Evaluation
+from autoblocks._impl.tracerv2.models import SendEventResponse
+from autoblocks._impl.util import AutoblocksEnvVar
+from autoblocks._impl.util import make_replay_headers
+
+log = logging.getLogger(__name__)
+
+# Globals
```

Comment:
we should add a new file in like `autoblocks/global_state.py` that manages these across the entire package, e.g. for tracer, prompt manager, testing suites, etc. i think they can all share the same event loop background thread and client